lib/adapters/leveldb/index.js

```diff
@@ -10,7 +10,16 @@ function requireLeveldown() {
   // de-optimalization
   try {
     originalLeveldown = require('leveldown');
-  } catch (e) {}
+  } catch (err) {
+      if (err && err.message && err.message.match('Module version mismatch')) {
+          throw new Error([
+            err.message,
+            'This generally implies that leveldown was built with a different',
+            'version of node than that which is running now.  You may try',
+            'fully removing and reinstalling PouchDB or leveldown to resolve.'
+          ].join(' '));
+      }
```

Comment:
Yeah it's true; that `try/catch` is pretty generic and will throw on just about any error. That said, I think it's less dangerous to have a general `try/catch` in this case, because I don't believe npm specifies exactly what the name of the error you get when a module fails to load (correct me if I'm wrong).

Also keep in mind that we make a distinction here between synchronous throwing and asynchronous callbacks; in general we prefer asynchronous callbacks because synchronous throwing might get swallowed in a promise chain.

For a concrete suggestion, let's just reword [the current error](https://github.com/pouchdb/pouchdb/blob/79785ee234b77af36137969b73f5d80dd7f0b0e4/lib/adapters/leveldb/index.js#L140-L147) to be your more helpful version. The `console.warn` stuff is unnecessary if users are already seeing the error.